aiidalab_launch/profile.py

```diff
@@ -106,6 +151,20 @@ def from_container(cls, container: Container) -> Profile:
             else container.image.tags[0]
         )
 
+        extra_destinations: list[PurePosixPath] = [
```

Comment:
This code flaky unfortunately, we're not guaranteed to get the extra mounts in the same order as they appeared in the original Profile. Because of this, the test in test_instance.py is flaky. https://github.com/aiidalab/aiidalab-launch/runs/7045495764?check_suite_focus=true

Not sure how to deal with this. Maybe we need to pass the expected extra_mounts as an optional parameter here to match the order. We could also specialize the `__eq__` method for Profile, but that seems like an awful lot of work.

Reply:
Okay, I think I solved by converting extra_mounts to Set(). Wasn't even that difficult a change, although I did have to patch the TOML loading.